src/modules/collection/sagas.ts

```diff
@@ -257,6 +257,24 @@ export function* collectionSaga(builder: BuilderAPI, catalyst: CatalystClient) {
         throw new Error(yield call(t, 'sagas.item.missing_salt'))
       }
 
+      // Check that items currently in the builder match the items the user wants to publish
+      // This will solve the issue were users could add items in different tabs and not see them in the tab
+      // were the publish is being made, leaving the collection in a corrupted state.
+      const serverItems: Item[] = yield call([builder, builder.fetchCollectionItems], collection.id)
+
+      if (serverItems.length !== items.length) {
+        throw new Error(`UnsyncedCollection: Different items length`)
+      }
+
+      // TODO: Deeper comparison of browser and server items. Compare metadata for example.
+      serverItems.forEach(serverItem => {
+        const browserItem = items.find(item => item.id === serverItem.id)
+
+        if (!browserItem) {
+          throw new Error(`UnsyncedCollection: No matching item in browser`)
```

Comment:
I would narrow the exception to just the existence of the item for now:
```suggestion
          throw new Error(`UnsyncedCollection: Item found in the server but not in the browser`)
```